File: OverlappingRectangles.py
class Point:
    # Lớp đại diện cho một điểm trong hệ tọa độ 2D
    def __init__(self, x, y):
        self.x = x  # Tọa độ x của điểm
        self.y = y  # Tọa độ y của điểm

def overlappingRectangles(l1, r1, l2, r2):
    # Nếu một hình chữ nhật nằm hoàn toàn bên trái hình kia
    if l1.x > r2.x or l2.x > r1.x: return False
    # Nếu một hình chữ nhật nằm hoàn toàn phía trên hình kia
    if r1.y < l2.y or r2.y < l1.y: return False
    # Nếu không rơi vào các trường hợp trên, hình chữ nhật giao nhau
    return True

if __name__ == "__main__":
    print("Nhập tọa độ cho hình chữ nhật 1:")
    l1_x, l1_y = map(int, input("Tọa độ góc trên trái (x, y): ").split())
    r1_x, r1_y = map(int, input("Tọa độ góc dưới phải (x, y): ").split())
    
    print("Nhập tọa độ cho hình chữ nhật 2:")
    l2_x, l2_y = map(int, input("Tọa độ góc trên trái (x, y): ").split())
    r2_x, r2_y = map(int, input("Tọa độ góc dưới phải (x, y): ").split())
    # Tạo các điểm
    l1, r1, l2, r2 = Point(l1_x, l1_y), Point(r1_x, r1_y), Point(l2_x, l2_y), Point(r2_x, r2_y)

    if overlappingRectangles(l1, r1, l2, r2):
        print("Hai hình chữ nhật giao nhau.")
    else:
        print("Hai hình chữ nhật không giao nhau.")

File: OverlappingRectangles.c
#include <stdio.h>
#include <stdbool.h>

bool overlappingRectangles(int l1x, int l1y, int r1x, int r1y, int l2x, int l2y, int r2x, int r2y){
    if (l1x > r2x || l2x > r1x) return false;
    if (l1y < r2y || l2y < r1y) return false;
    // Nếu không thuộc hai trường hợp trên, chúng chồng lên nhau
    return true;
}
int main(){
    int l1x, l1y, r1x, r1y, l2x, l2y, r2x, r2y;

    printf("Hãy nhập tọa độ điểm l1 (góc trên trái) và r1 (góc dưới phải) của HCN thứ 1:\n");
    printf("Tọa độ l1x l1y: "); scanf("%d %d", &l1x, &l1y);
    printf("Tọa độ r1x r1y: "); scanf("%d %d", &r1x, &r1y);

    printf("\nHãy nhập tọa độ điểm l2 (góc trên trái) và r2 (góc dưới phải) của HCN thứ 2:\n");
    printf("Tọa độ l2x l2y: "); scanf("%d %d", &l2x, &l2y);
    printf("Tọa độ r2x r2y: "); scanf("%d %d", &r2x, &r2y);

    if (overlappingRectangles(l1x, l1y, r1x, r1y, l2x, l2y, r2x, r2y)) {
        printf("\nHai hình chữ nhật chồng lên nhau\n");
    } else {
        printf("\nHai hình chữ nhật không chồng lên nhau\n");
    }

    return 0;
}
